src/Reflection/ReflectionClass.php

```diff
@@ -370,22 +380,17 @@ public function getEndLine()
      * You may optionally specify a source locator that will be used to locate
      * the parent class. If no source locator is given, a default will be used.
      *
-     * @param SourceLocator|null $sourceLocator
      * @return ReflectionClass
      */
-    public function getParentClass(SourceLocator $sourceLocator = null)
+    public function getParentClass()
     {
         if (null === $this->extendsClassType) {
             return null;
         }
 
-        $fqsen = $this->extendsClassType->__toString();
-
-        if (null !== $sourceLocator) {
-            $parent = (new ClassReflector($sourceLocator))->reflect($fqsen);
-        } else {
-            $parent = self::createFromName($fqsen);
-        }
+        // @TODO use actual `ClassReflector` or `FunctionReflector`?
```

Comment:
I am still not sure, that's why it's still there. I have no specific `ClassReflector` hinting, therefore the usage is unsafe in here